Show reference actions in new UX

umpox opened the pull request:
<!-- Describe the purpose of the PR so that if you looked at it in 6 months time it would be clear from the overview why this was created -->
## Overview
This PR moves the reference action items to the top bar, and removes their min width.

Note: There is some tricky styles to implement here. The design indicates that we want these action items (e.g. `Group by file`) to break onto a new line responsively. I didn't want to implement this using just a media query, because the length and amount of these items are unknown (populated by extensions). Now we just use `flex-wrap` to ensure content is pushed onto a new line, and we use a psuedo element to modify the `border-bottom` of the reference items to always fill the full width of the tab bar.

<!-- Insert before and after screenshot(s) that makes it easy to identify the changes made -->
## Screenshots desktop
| BEFORE             |  AFTER |
:-------------------------:|:-------------------------:
![image](https://user-images.githubusercontent.com/9516420/108703574-b57a2f00-7502-11eb-974a-30f9dee439a2.png) | ![image](https://user-images.githubusercontent.com/9516420/108703607-c034c400-7502-11eb-9a28-6f27e490c927.png)

## Screenshots mobile
| BEFORE             |  AFTER |
:-------------------------:|:-------------------------:
![image](https://user-images.githubusercontent.com/9516420/108703685-dc386580-7502-11eb-8baa-f4a4ca8227e1.png) | ![image](https://user-images.githubusercontent.com/9516420/108703712-e5293700-7502-11eb-95cf-abdabff6b38b.png)


<!-- Update the implementation steps that should be completed to implement the feature/bug fix/tech debt cleanup -->
## Progress
- [X] Documentation text/screenshots updated (if relevant)
- [x] Approved by a frontend engineer (if touching frontend code)
- [ ] Approved by a designer (if it touches the UI)

<!-- Add a reference to the issue that this PR addresses if relevant -->
Closes https://github.com/sourcegraph/sourcegraph/issues/9740

AlicjaSuska commented:
Hi! Thank you for sharing this work. As we've discussed: the only thing that is cannot be aligned with the design is the thickness of the bottom line on mobile. I'm ok with it for now, it's not a blocker for me, especially if it's time-consuming to change.

Design vs implementation:
<img width="918" alt="Screenshot 2021-02-23 at 12 07 34" src="https://user-images.githubusercontent.com/20326070/108840083-263e4b80-75d6-11eb-88d5-131660f119ab.png">
<img width="1023" alt="Screenshot 2021-02-23 at 12 08 00" src="https://user-images.githubusercontent.com/20326070/108840089-28080f00-75d6-11eb-90fb-4d5f5a2d40ac.png">

Other notes from me: please, keep the text styles of action items the same as in current implementation: text color and text size should be different (is: 14px, should be 12px). It looks like something is overwriting the link color (contrast is too low both in light and dark theme). thank you!

AlicjaSuska reviewed (changes requested):
Please, make changes to the text styles (I've listed them in my previous comment). Thank you!

umpox commented:
@AlicjaSuska 

I believe I've made these changes, the font sizes match the designs unless I'm missing something there? Updated example (I changed the text temporarily here to match the designs)
![image](https://user-images.githubusercontent.com/9516420/108868186-dc615f80-75ed-11eb-8ebe-cf748f9a10c1.png)

umpox commented on client/branded/src/components/panel/Panel.scss:
```diff
@@ -50,4 +57,33 @@
+        padding-right: 2rem;
+
+        // Ensures the action border always completes the full line it breaks onto
+        &::after {
```
For reviewer:

This is the cleanest way I can think of reliably achieving this design, we essentially want to break these actions onto a new line depending on their length. This means these actions have no way of understanding when they are in a 'new row' (without using JS), so in this scenario we ensure the `border-bottom` always covers the full row, despite its positioning.

Interested to hear if you have any other solutions

felixfbecker replied:
Do you have a screenshot of when they are wrapped (i.e. the desired visuals)?

umpox replied:
These are the designs https://www.figma.com/file/yPrTtrvmkUtizDuzWiqzrZ/9740-ActionItems-in-references-panel-don-t-look-good-Approved?node-id=246%3A11

Screenshots of the implementation are in the PR description, unless you mean something else?

felixfbecker commented:
Would be great to have a story in Storybook to try out :)

tjkandala reviewed (approved):
The CSS solution seems fine and I definitely prefer this over a JS solution here. I'm actually doing something similar for the action items bar :) LGTM for frontend code.

Visually, action items don't look vertically centered while my expectation would be that they are (On Linux Chromium | 88.0.4324.182 (Official Build) snap (64-bit))

![Screenshot from 2021-02-24 12-33-11](https://user-images.githubusercontent.com/37420160/109042438-33267200-769e-11eb-9711-14be82582fb8.png)
![Screenshot from 2021-02-24 12-32-44](https://user-images.githubusercontent.com/37420160/109042448-34f03580-769e-11eb-93d9-42efd551099a.png)

AlicjaSuska reviewed (changes requested):
Thank you for updating the labels @umpox In the design, font size was 12px (as it was in the original implementation). Here, it's 14px. Now, that I look at it, it's better this way. Let's leave it :)

There is only one small change I'd like to ask for: we've shortened the label to 'Hide / show other repositiories' not 'Hide / show ~~references from~~ other repositories'.

With this change, it's ready to go! 🎉

umpox commented:
Thanks @AlicjaSuska, the copy change is from code in a different repo, so I have made a separate PR there: https://github.com/sourcegraph/code-intel-extensions/pull/579

AlicjaSuska reviewed (approved):
Thank you for making changes to the copy in a separate PR @umpox. In this case, all good to go! 🎉
